lib/sdk.dart

```diff
@@ -292,6 +290,15 @@ class SdkDiscovery {
     }
   }
 
+  /// Return the path to the Dart sdk contained within the Flutter sdk, if
+  /// either exist.
+  String _tryFlutterSdk() {
+    if (!_flutterSdkManager.hasSdk) return null;
+    return _flutterSdkManager.sdk.dartSdkPath;
+  }
+
+  FlutterSdkManager get _flutterSdkManager => deps[FlutterSdkManager];
+
   String _resolveSdkFromVm(String vmPath) {
     if (vmPath == null) return null;
```

Comment:
`which` will throw if it can't locate the executable - it'll always return a non-null String. The null check is a bit defensive, but it makes sense to add an empty string check; will do.